core/src/HIP/Kokkos_HIP_Space.hpp

```diff
@@ -77,15 +77,9 @@ class HIPSpace {
     return allocate(arg_label, arg_alloc_size, arg_logical_size);
   }
 #else
-  // FIXME_HIP Use execution space instance
-  void* allocate(const HIP&, const size_t arg_alloc_size) const {
-    return allocate(arg_alloc_size);
-  }
-  // FIXME_HIP Use execution space instance
+  void* allocate(const HIP&, const size_t arg_alloc_size) const;
```

Comment:
```suggestion
  void* allocate(const HIP& exec_space, const size_t arg_alloc_size) const;
```
Should we include the name since the function impl is using `exec_space`? To be consistent with `arg_alloc_size`.